src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1586,6 +1586,42 @@ && getOwnBuilding().getCompletedRequestsOfTypeFiltered(worker.getCitizenData(),
         return false;
     }
 
+    /**
+     * Check if a request exists for a given deliverable.
+     * @param deliverable the deliverable to check of request.
+     * @return true if available or transfered.
+     */
+    public boolean checkIfRequestForItemExistOrCreate(@NotNull final IDeliverable deliverable)
+    {
+        final int invCount = InventoryUtils.getItemCountInItemHandler(worker.getInventoryCitizen(), deliverable::matches);
```

Comment:
we had some better inv util for cases like these where you do not want to know the full count, but if it has at least X as that allows early stops

Reply:
We actually do want to know the full count so that we can calculate the difference